### PR TITLE
[WIP] Check return types/API conformity

### PR DESCRIPTION
**Pull request by betatim**
Start checking that we stick to the correct return types. Where have we got defined APIs?

`Space().rvs()` -> `list of lists`. The `*_minimize()` functions return a result object.

### CODE REVIEW

**Comment by MechCoder** [comment]
LGTM. Can you add a test for `inverse_transform` as well?

**Comment by MechCoder** [comment]
Would you like to test the return type of the result objects in this PR as well? In other words before or after https://github.com/scikit-optimize/scikit-optimize/pull/119/ ?

**Comment by MechCoder** [comment]
Here are the list of stuff, obtained from `gp_minmize`

`x` [float]: location of the minimum.
`fun` [float]: function value at the minimum.
 `models`: surrogate models used for each iteration.
 `x_iters` [list of lists]
`func_vals` [list]: function value for each iteration.
`[Space]`: the optimization space.

Minor: Should we relax the condition on `func_vals` ti be an ndarray instead of list?

**Comment by betatim** [comment]
After #119

**Comment by MechCoder** [comment]
ok, then I'll merge after the inverse_transform check. Thanks!

**Comment by betatim** [comment]
make a common test out of the separate tests for trees and GPs (see #146)

File: skopt/tests/test_common.py [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+from sklearn.utils.testing import assert_almost_equal
+from sklearn.utils.testing import assert_array_less
+from sklearn.utils.testing import assert_array_equal
+from sklearn.utils.testing import assert_equal
+from sklearn.utils.testing import assert_raises
+
+from skopt.benchmarks import branin
+from skopt.dummy_opt import dummy_minimize
+from skopt.gp_opt import gp_minimize
+from skopt.tests.test_tree_opt import MINIMIZERS as TREE_MINIMIZERS
```
Comment (betatim) [inline skopt/tests/test_common.py]: How can I make this work in legacy python? Alternatively should there be `skopt.utils` with a `all_minimizers()`? Trying to find a nice way to not repeat `MINIMIZERS` from the tree tests.

Comment (glouppe) [inline skopt/tests/test_common.py]: I would just repeat it for now. Then maybe we could factor this out in `skopt.testing` util.

**Comment by betatim** [comment]
Anymore for anymore?

**Comment by MechCoder** [comment]
thanks!

**Comment by betatim** [comment]
Bim! 🏁

Happy weekend y'all!

**Pull request merged**